### PR TITLE
Add Dynamic DNS setup instructions

### PR DESCRIPTION
**Pull request by NicolasDorier**

### CODE REVIEW

File: DynamicDNS.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Dynamic DNS Service
+
+## Motivation
+
+You **need** Dynamic DNS Service if:
```
Comment (pavlenex) [inline DynamicDNS.md]:
```suggestion
Dynamic DNS is needed if:
```

Comment (pavlenex) [reply DynamicDNS.md]: Just a bit more formal ^

File: DynamicDNS.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+1. Install BTCPay Server on a host provider which do not provide a default domain for your server,
+2. Don't want to buy your own domain name (ie, `mybusiness.com`)
```
Comment (pavlenex) [inline DynamicDNS.md]:
```suggestion
2. You don't want to buy your own domain name (ie, `mybusiness.com`)
```

File: DynamicDNS.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+3. Need to access your BTCPay Server over internet via HTTPS. (Your BTCPayServer is accessed by other internet users)
+
+Or
+1. You host BTCPayServer at home and are not guarantee to by your internet service provider to keep the same IP
```
Comment (pavlenex) [inline DynamicDNS.md]:
```suggestion
1. You host BTCPayServer at home and your internet service provider does not guarantee you the same IP
```

File: DynamicDNS.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Dynamic DNS Service
+
+## Motivation
+
+You **need** Dynamic DNS Service if:
+1. Install BTCPay Server on a host provider which do not provide a default domain for your server,
```
Comment (pavlenex) [inline DynamicDNS.md]:
```suggestion
1. BTCPay Server is installed a hosting provider which does not provide a default domain for your server.
```

File: DynamicDNS.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+Then you want to use BTCPayServer Dynamic DNS service.
+
+You **don't need** Dynamic DNS Service if:
+
+1. You host BTCPayServer at home and only access it via the local network (just using HTTP locally or using Tor is fine)
+2. Your BTCPayServer should be only accessed by yourself (Just use Tor browser and the Tor address of you instance)
+3. Your hosting provider provide some domain name for your server by default (ie, Lunanode for example provide a subdomain of `.lndyn.com` for free, and Azure provides for `.azurewebsites.net`)
+
+A Dynamic DNS Providers allows you to have a free domain like `example.ddns.net` for your server, additionally Dynamic DNS Providers exposes a simple API to update the DNS record automatically when your BTCPayServer instance change it's external IP address.
```
Comment (pavlenex) [inline DynamicDNS.md]:
```suggestion
A Dynamic DNS Providers allows you to have a free domain like `example.ddns.net` for your server, additionally Dynamic DNS Providers exposes a simple API to update the DNS record automatically when your BTCPayServer instance change its external IP address.
```

File: DynamicDNS.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+BTCPayServer, when configured to use Dynamic DNS, will periodically check and update the DNS record if an external IP change is detected.
+
+## How to use
+
+### Step 1: Create the domain
+You need first to create an account on one Dynamic DNS provider, the most popular are:
```
Comment (pavlenex) [inline DynamicDNS.md]:
```suggestion
First, create an account on a Dynamic DNS provider, the most popular providers are:
```

**Comment by NicolasDorier** [comment]
If you change file names, let me know because it breaks links in the app.

**Comment by pavlenex** [comment]
No it does not change file name, it will just change summary.md which dictates what's displayed on GitBook.

**Pull request merged**